services/kvstore/src/main/java/tech/pegasys/pantheon/services/kvstore/KeyValueStorage.java

```diff
@@ -35,6 +35,14 @@ default boolean containsKey(final BytesValue key) throws StorageException {
    */
   Optional<BytesValue> get(BytesValue key) throws StorageException;
 
+  /**
+   * Removes a key/value pair from storage.
+   *
+   * @param key Index into storage
+   * @return 1 if a key was removed, 0 otherwise
+   */
+  long remove(BytesValue key) throws StorageException;
```

Comment:
If you're modifying this interface, you can quickly add tests across the board by updating `AbstractKeyValueStorageTest`.  But not sure if we actually need this new method, since the `Transaction` already supports removals ...

Reply:
Hmm, ok, I'm leaning towards just using the `Transaction` from `KeyValueStorage` but it looks like `WorldStateKeyValueStorage` doesn't implement `KeyValueStorage`, was this an intentional decision?

Reply:
Hmm, ok, I'm leaning towards just using the `Transaction` from `KeyValueStorage` but it looks like `WorldStateKeyValueStorage` doesn't implement `KeyValueStorage`, was this an intentional decision?

Reply:
`KeyValueStorage` is a layer below `WorldStateKeyValueStorage` (this class is implemented using a `KeyValueStorage`).  I think you'd need to add a method like `removeAccountStateTrieNode` to the `Updater`.